### PR TITLE
Add new vehicle Mercedes Smart ED

### PR DESCRIPTION
**Pull request by martingraml**
Add basic functionality for Mercedes Smart electric drive.

### CODE REVIEW

File: vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+#include "ovms_peripherals.h"
+
+OvmsVehicleSmartED::OvmsVehicleSmartED() {
+	ESP_LOGI(TAG, "Start Smart ED vehicle module");
+
+	RegisterCanBus(1, CAN_MODE_ACTIVE, CAN_SPEED_500KBPS);
+
+	// require GPS:
+	MyEvents.SignalEvent("vehicle.require.gps", NULL);
+	MyEvents.SignalEvent("vehicle.require.gpstime", NULL);
```
[the remark this answers is not in the document]
Comment (martingraml) [reply vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp]: I took these commands from another vehicle, i think Mitsubishi.
So i can remove these commands?

File: vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+
+void OvmsVehicleSmartED::Ticker60(uint32_t ticker) {
+
+}
+
+
+OvmsVehicle::vehicle_command_t OvmsVehicleSmartED::CommandClimateControl(
+		bool enable) {
+	OvmsVehicleSmartED::CommandSetChargeTimer(enable, StandardMetrics.ms_m_timeutc->AsInt(0, Seconds));
```
[the remark this answers is not in the document]
Comment (martingraml) [reply vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp]: Yes this is for the passenger cabin. For the Smart ED it si special, you have to set the drive away time and 20 minutes before the air condition starts. This is the same as the charge timer.
This car doesn´t really have a charge timer it starts charging immediately so you have to set the charge timer to condition the cabin.

File: vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+	return Success;
+}
+
+void SmartEDLockingTimer(TimerHandle_t timer)
+  {
+    xTimerStop(timer, 0);
+    xTimerDelete(timer, 0);
+	//reset GEP 1 + 2
+	MyPeripherals->m_max7317->Output(MAX7317_EGPIO_1, 0);
```
[the remark this answers is not in the document]
Comment (martingraml) [reply vehicle/OVMS.V3/components/vehicle_smarted/src/vehicle_smarted.cpp]: is there an alternative for the max7317 if they are not in future builds?
They stay on the hardware?

File: vehicle/OVMS.V3/main/Kconfig [hunk 4]
```diff
@@ -92,18 +92,6 @@ config OVMS_HW_CAN_TX_QUEUE_SIZE
...
-
-config OVMS_SYS_COMMAND_STACK_SIZE
-    int "Stack size for command execution"
-    default 6144
-    depends on OVMS
-    help
-        The stack size of the OVMS Console and dynamic command tasks.
-
-endmenu # System Options
-
```
[the remark this answers is not in the document]
Comment (M4GNV5) [reply vehicle/OVMS.V3/main/Kconfig]: This doesn't really seem to be, but is marked as resolved.

**Comment by M4GNV5** [comment]
Apart from the things @dexterbg already said and the mixed german comments it looks fine. A couple of CAN messages are labelled but ignored and some metric sets are commented out, I assume there were some issues with them?

What Smart ED version did you use to try this? Do you know if there is a difference between the [old](https://www.evilution.co.uk/files/info/ed/ed.jpg) and [new](http://cdn1.carbuyer.co.uk/sites/carbuyer_d7/files/car_images/smart-fortwo-ed.jpg) models apart from the look?

**Pull request closed without merging**